state-chain/runtime/src/lib.rs

```diff
@@ -2334,40 +2343,108 @@ impl_runtime_apis! {
 			}
 		}
 
-		fn cf_get_open_deposit_channels(account_id: Option<AccountId>) -> ChainAccounts {
-			let btc_chain_accounts = pallet_cf_ingress_egress::DepositChannelLookup::<Runtime,BitcoinInstance>::iter_values()
-				.filter(|channel_details| account_id.is_none() || Some(&channel_details.owner) == account_id.as_ref())
-				.map(|channel_details|
-					channel_details.deposit_channel.address
-					.into_foreign_chain_address()
-					.to_encoded_address(Environment::network_environment())
-				)
-				.collect::<Vec<_>>();
+		fn cf_get_open_deposit_channels(account_id: Option<<Runtime as frame_system::Config>::AccountId>) -> ChainAccounts {
+			fn open_deposit_channels_for_account<T: pallet_cf_ingress_egress::Config<I>, I: 'static>(
+				account_id: Option<&<T as frame_system::Config>::AccountId>
+			) -> Vec<EncodedAddress>
+			{
+				let network_environment = Environment::network_environment();
+				pallet_cf_ingress_egress::DepositChannelLookup::<T, I>::iter_values()
+					.filter(|channel_details| account_id.is_none() || Some(&channel_details.owner) == account_id)
+					.map(|channel_details|
+						channel_details.deposit_channel.address
+							.into_foreign_chain_address()
+							.to_encoded_address(network_environment)
+					)
+					.collect::<Vec<_>>()
+			}
 
 			ChainAccounts {
-				chain_accounts: btc_chain_accounts
+				chain_accounts: [
+					open_deposit_channels_for_account::<Runtime, BitcoinInstance>(account_id.as_ref()),
+					open_deposit_channels_for_account::<Runtime, EthereumInstance>(account_id.as_ref()),
+				].into_iter().flatten().collect()
 			}
 		}
 
+		fn cf_all_open_deposit_channels() -> Vec<(AccountId, ChannelActionType, ChainAccounts)> {
+			use sp_std::collections::btree_set::BTreeSet;
+
+			#[allow(clippy::type_complexity)]
+			fn open_deposit_channels_for_chain_instance<T: pallet_cf_ingress_egress::Config<I>, I: 'static>()
+				-> BTreeMap<(<T as frame_system::Config>::AccountId, ChannelActionType), Vec<EncodedAddress>>
+			{
+				let network_environment = Environment::network_environment();
+				pallet_cf_ingress_egress::DepositChannelLookup::<T, I>::iter_values()
+					.fold(BTreeMap::new(), |mut acc, channel_details| {
+						acc.entry((channel_details.owner.clone(), channel_details.action.into()))
+							.or_default()
+							.push(
+								channel_details.deposit_channel.address
+								.into_foreign_chain_address()
+								.to_encoded_address(network_environment)
+							);
+						acc
+					})
+			}
+
+			let btc_chain_accounts = open_deposit_channels_for_chain_instance::<Runtime, BitcoinInstance>();
+			let eth_chain_accounts = open_deposit_channels_for_chain_instance::<Runtime, EthereumInstance>();
+			let accounts = btc_chain_accounts.keys().chain(eth_chain_accounts.keys()).cloned().collect::<BTreeSet<_>>();
```

Comment:
This should include `arb_chain_accounts` AND they should be added to the `accounts` chaining.